src/monodroid/jni/android-system.cc

```diff
@@ -616,22 +627,10 @@ AndroidSystem::get_existing_dso_path_on_disk (const char *base_dir, const char *
 		return dso_path;
 
 	*needs_free = false;
-	free (dso_path);
+	delete[] dso_path;
 	return nullptr;
 }
 
-void
-AndroidSystem::dso_alloc_cleanup (char **dso_path, bool *needs_free)
-{
-	assert (needs_free != nullptr);
-	if (dso_path != nullptr) {
-		if (*needs_free)
-			free (*dso_path);
-		*dso_path = nullptr;
-	}
-	*needs_free = false;
-}
-
 char*
 AndroidSystem::get_full_dso_path_on_disk (const char *dso_name, bool *needs_free)
 {
```

Comment:
Is the `needs_free` parameter still needed here? Looks like it is only called from `setup_bundled_app`, where it is not used.

Reply:
Good call, I'll review the code.

Reply:
We use it, [see here](https://github.com/xamarin/xamarin-android/pull/3500/commits/b03dbd43858ea2ff44d63af467efc5f7a32c9e55#diff-09282c2637d665875eb937b5222b94f7R235)

Reply:
The link is not working anymore

Reply:
OK, I see it is now used in `setup_bundled_app`